docs/tutorials/podman_pasta_ipv6_with_snid_ipv4.md

```diff
@@ -0,0 +1,761 @@
+# - https://www.agwa.name/blog/post/using_sni_proxying_and_ipv6_to_share_port_443
+# - https://www.rfc-editor.org/rfc/rfc6052
+# - https://github.com/luckylinux/ipv6-decode-ipv4-address
+
+[Unit]
+Description=SNID Service
+
+[Service]
+# Running as rootless does NOT appear to work, even when adding AmbientCapabilities=CAP_NET_BIND_SERVICE
+User=root
+ExecStartPre=/bin/bash -c 'sleep 10'
+ExecStart=/bin/bash -c 'cd /opt/snid && ip route replace local 64:ff9b:1::/96 dev lo && ./snid -listen tcp:172.16.1.10:443 -mode nat46 -nat46-prefix 64:ff9b:1:: -backend-cidr 2001:db8:0000:0001:0000:0000:0001:0001/112'
+ExecStop=/bin/bash -c 'cd /opt/snid && ip route del local 64:ff9b:1::/96 dev lo'
+
+[Install]
+WantedBy=multi-user.target
+```
+
+Reload Systemd, enable and Start the Service:
+```
+systemctl daemon-reload
+systemctl enable snid.service
+systemctl restart snid.service
+systemctl status snid.service
+```
+
+Check that no Errors occurred !
+
+## Rootless Service
+An enhancement of the original `/etc/systemd/system/snid.service` can be done, by splitting up the Service in two Parts:
+- Routes Setup Service (requires Root Privileges): `/etc/systemd/system/snid-routes.service`
+- Server Service (can be run as Normal User): `/etc/systemd/system/snid-server.service`
+
+Since the Routes are Setup in less than a Second and, after that, `snid` is run as Normal User, the attack Surface is minimized. Even better would be to run the Service as a User which is DIFFERENT from the User Running `podman` (to ensure that the Files owned by `podman` CANNOT be accessed in case `snid` is compromized).
+
+Of course this requires (but should be already in Place, since the Rootless Containers require it already) in `/etc/sysctl.conf`:
+```
+net.ipv4.ip_unprivileged_port_start=80
+net.ipv4.ping_group_range=0 2000000
+kernel.unprivileged_userns_clone=1
+
+net.ipv6.ip_nonlocal_bind=1
+```
+
+Furthermore concerning the User Account Setup (it is preferable to use a different User compared to the one running the Podman Containers, for the Reasons outlined above):
+```
+groupadd snid
+useradd --shell /usr/sbin/nologin -g snid --base-dir /opt/snid snid
+chown snid:snid /opt/snid/snid
+chmod 0770 /opt/snid/snid
+```
+
+`/etc/systemd/system/snid-routes.service`:
+```
+# To get SNID to work:
+# - ip route add local the_nat46_prefix/96 dev lo
+# - nmcli connection modify lo +ipv6.routes "64:ff9b:1::/96 dev lo" -> NOT WORKING
+# - Backend CIDR: 2001:db8:0000:0001:0000:0000:0001:0001/112 (2001:db8:0000:0001:0000:0000:0001:0000 ... 2001:db8:0000:0001:0000:0000:0001:ffff)
+#
+# Convert IPv4 Address to IPv6 Address Representation: 
+# - https://www.agwa.name/blog/post/using_sni_proxying_and_ipv6_to_share_port_443
+# - https://www.rfc-editor.org/rfc/rfc6052
+
+[Unit]
+Description=SNID Routes Setup Service
+
+[Service]
+#Type=oneshot
+User=root
+ExecStart=/bin/bash -c 'ip route replace local 64:ff9b:1::/96 dev lo'
+ExecStop=/bin/bash -c 'ip route del local 64:ff9b:1::/96 dev lo'
```

Comment:
drop `bash -c`

Reply:
Are you sure ??? 

I'm pretty sure I had lots of issues (cannot remember if due to missing ENV Variables or when running complex commands i.e. with && or with more than one Argument, but I had lots of Issues when NOT using `/bin/bash -c` as a "Starter").

That's why I have been using it ever since.

Reply:
It's really not necessary here. You are not making use of any shell functionality. It's just a static command.

Reply:
OK fair enough, I just tested it and it works as you say :+1:.

Reply:
Fixed.